hooks/image/53-ek-content-preload

```diff
@@ -14,11 +14,13 @@ channels_file="${EIB_TMPDIR}"/ek-channels
 rm -f "${channels_file}"
 touch "${channels_file}"
 all_collection_files="${OSTREE_VAR}"/lib/flatpak/app/org.endlessos.Key/current/active/files/share/endless-key/collections/*.json
+# Convert whitespace-separated list of collections to a colon-delimited string
+requested_collections=":$(echo -n "${EIB_ENDLESSKEY_COLLECTIONS}" | tr -s '[:space:]' ':'):"
 for collection_file in ${all_collection_files}; do
   # Check if the file basename stripped off of -0001.json is part of the list
   # of collections to be installed.
-  bn=$(basename ${collection_file})
-  if [[ "${EIB_ENDLESSKEY_COLLECTIONS}" =~ .*"${bn%-????.json}".* ]] ; then
+  bn=$(basename "${collection_file%-????.json}")
+  if [[ ${requested_collections} =~ :${bn}: ]] ; then
```

Comment:
Switching to `:` delimited is reasonable. However, here's an alternative that I believe does the right thing:

```
[[ ${EIB_ENDLESSKEY_COLLECTIONS} =~ (^|[[:space:]])${bn}($|[[:space:]]) ]]
```
or
```
[[ " ${EIB_ENDLESSKEY_COLLECTIONS} " =~ [[:space:]]${bn}[[:space:]] ]]
```

These would be simpler with the `\b` boundary class available in `grep` and perl REs.